src/components/PropertyCard/PropertyCard.js

```diff
@@ -14,9 +15,56 @@ import {
 import FavoriteIcon from '@mui/icons-material/Favorite';
 import ShareIcon from '@mui/icons-material/Share';
 
+import { useAuthUser } from '@/firebase/auth';
+import { getUser } from '@/firebase/userDb';
+import {
+	addUserPropertyPreference,
+	getUserListFromPropertyID,
+	getPropertyListFromUserID,
+	deleteUserPropertyMapping,
+} from '@/firebase/userLikedProperties';
+
 export default function PropertyCard({ property }) {
+	const [user, loading] = useAuthUser();
+	const [like, setLike] = useState(false);
+	const [likeUsers, setLikeUsers] = [];
+	const [avatarUrls, setAvatarUrls] = useState([]);
+
+	useEffect(() => {
+		if (user) {
+			getPropertyListFromUserID(user.uid, property.uid)
+				.then((res) => {
+					setLike(res.includes(property.uid));
+				})
+				.catch((err) => {
+					console.log(err);
+				});
+
+			getUserListFromPropertyID(property.uid)
+				.then((res) => {
+					let urls = [];
+					for (const uid of res) {
+						getUser(uid).then((u) => {
+							urls.push(u.imageUrl);
+							setAvatarUrls(urls);
+						});
+					}
+				})
+				.catch((err) => console.log(err));
+		}
+	}, [user, property.uid]);
+
+	const handleLike = (e) => {
+		setLike(!like);
+		if (like) {
+			deleteUserPropertyMapping(user.uid, property.uid);
+		} else {
+			addUserPropertyPreference(user.uid, property.uid);
+		}
```

Comment:
This section took me a little bit to parse.

First, we set the like to the opposite - that makes total sense, the button was just clicked, we want to flip it.

But then we kind of do opposite logic - `if (like)` really means "the button *was* liked, now we unliked it, delete the mapping". When quickly scanning the code, it seems backwards - we liked the property, so delete the mapping? It takes understanding to get that `like` was the PREVIOUS state, not the state right now.
 
Can it instead be something like
`
let like = !like // flip the like
`
and then just use `like`  everywhere? (so `setLike(like)` to the new status and make `if (like)` add the property pref)

Reply:
Got it. Fixed.